src/reconcile/rec_visibility.c

```diff
@@ -117,18 +121,16 @@ __rec_append_orig_value(WT_SESSION_IMPL *session, WT_PAGE *page, WT_INSERT *ins,
      * Else, set the entry's transaction information to the lowest possible value. Cleared memory
      * matches the lowest possible transaction ID and timestamp, do nothing.
      */
-    if (upd != NULL && upd->type == WT_UPDATE_BIRTHMARK) {
-        append->txnid = upd->txnid;
-        append->start_ts = upd->start_ts;
-        append->durable_ts = upd->durable_ts;
+    if (upd->type == WT_UPDATE_BIRTHMARK)
         append->next = upd->next;
-    }
+
+    if (tombstone != NULL)
+        append = tombstone;
 
     /* Append the new entry into the update list. */
-    if (upd_entry != NULL)
-        WT_PUBLISH(*upd_entry, append);
-    else
-        WT_PUBLISH(upd->next, append);
+    WT_PUBLISH(upd->next, append);
+    /* Timestamp should always be in descending order */
+    WT_ASSERT(session, upd->start_ts >= append->start_ts);
     __wt_cache_page_inmem_incr(session, page, size);
```

Comment:
I'd suggest moving this line below the code to reset the `upd` type. It doesn't need to be in the middle of stuff, it's kind of an after-thought.

Reply:
Thanks, I didn't realized I need to take care of the size and I mishandled it completely. Fixed now.